### DISCUSSION TITLE
Python scripting clearListing / createData() issues

### DISCUSSION THREAD

**Discussion by phix33** [NationalSecurityAgency/ghidra#6247] [Q&A] [answered]
Really useful discussion at https://github.com/NationalSecurityAgency/ghidra/discussions/4862

However, I'm still having issues.....when analysing a 37Mb firmware binary for structures, I'm trying to create data structures as follows:

```
    tlen = mydata_t.clone(currentProgram.getDataTypeManager()).getLength()
    fpapi.clearListing(addr, addr.add(tlen - 1))
    fpapi.createData(addr, mydata_t)
```

However, I routinely get `CodeUnitInsertionException: Data conflict at...`. Now I sort of assumed that the clearListing would have removed the data at the given address but when it fails I can see that the address has a pointer defined rather than the data structure however if I clear the given address and re-run the script I do seem to get further ... so, is there a better way of clearing/removing previously defined data?

[ I even tried running various clearListing with varying ranges and even just clearListing(addr) before the createData but no "luck". ]

As suggested in the original discussion, I checked the addresses are always part of the memory space (as I can jump to the address and see it). 

Finally, after various iterations the script _does_ successfully run to completion. However, I am on the verge of running out of memory too - I had to change MAXMEM to 30G (on 32G) laptop. The script isn't threaded  and is "just" iterating over a list of structure addresses.

**Comment by mumbel** [comment, marked as answer]
Maybe a synchronization issue with python?  I have run into this as well (Can't remember if I've made an issue or not, or ever brought it up in related issues).  I just make a wrapper/helper function with the address and the datatype and use that instead, have not had it fail twice with the second in an `except` and a minor yield

```pytyon
try:
    clearListing(addr, addr.add(tlen - 1))
    createData(addr, mydata_t)
except:
    time.sleep(0.01)
    clearListing(addr, addr.add(tlen - 1))
    createData(addr, mydata_t)   
```

**Comment by phix33** [comment]
Nice, that worked!!
